libs/langchain/langchain/chains/hyde/base.py

```diff
@@ -72,14 +74,28 @@ def from_llm(
         cls,
         llm: BaseLanguageModel,
         base_embeddings: Embeddings,
-        prompt_key: str,
+        prompt_key: Optional[str] = None,
+        custom_prompt: Optional[PromptTemplate] = None,
+        run_manager: Optional[CallbackManagerForChainRun] = None,
```

Comment:
run_manager creation is handled in background and shouldn't be instantiated / passed around by users. you'd want to pass in `Callbacks` but that can already be down via **kwargs

Reply:
I think it's not possible right now, because of the config: 
    class Config:
        """Configuration for this pydantic object."""

        extra = Extra.forbid
        arbitrary_types_allowed = True

Reply:
the class inherits from Chain which has an explicit `Chain.callbacks` parameter, so i think it doesn't count as an extra

Reply:
I didn't see that! Thanks for pointing it out